sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPumpManager.java

```diff
@@ -244,8 +264,13 @@ void startPartitionPump(PartitionOwnership claimedOwnership, Checkpoint checkpoi
                 });
 
             if (maxWaitTime != null) {
-                partitionEventFlux = receiver
-                    .windowTimeout(maxBatchSize, maxWaitTime);
+                if (HAS_BACKPRESSURE_WINDOW_TIMEOUT) {
+                    partitionEventFlux = receiver
+                        .windowTimeout(maxBatchSize, maxWaitTime, true);
+                } else {
+                    partitionEventFlux = receiver
+                        .windowTimeout(maxBatchSize, maxWaitTime);
+                }
```

Comment:
I was trying to see if the direct reference (i.e. non-reflective reference) to the API here is safe? so far, my understanding is, the module gets compiled only in SDK repro and always with the reactor-core3.4.19 or above (i.e. new method is available at compile time).

Anyone else will be using the published compiled jar (e.g., from maven), and invocation of one of the method overloads is protected by its discoverability at runtime, i.e., code won't try to load the new API with the older reactor-core.

But want to hear from others who have more expertise in doing this.

Reply:
I found a somewhat similar direct reference example here https://github.com/netty/netty/blob/4.1/codec/src/main/java/io/netty/handler/codec/compression/JdkZlibEncoder.java#L334-L336

```java
private void deflate(ByteBuf out) {
    if (PlatformDependent.javaVersion() < 7) {
        deflateJdk6(out);
    }
    int numBytes;
    do {
       ...
        numBytes = deflater.deflate(
                out.array(), out.arrayOffset() + writerIndex, 
                out.writableBytes(), Deflater.SYNC_FLUSH);
       ...
    } while (numBytes > 0);
}
```

The `Deflater.SYNC_FLUSH` is directly referenced; the enum value "SYNC_FLUSH" available only from Java7, and runtime access to the value is protected by a similar runtime check.

Reply:
I have a little bit concern for the checking. Because we may already or in the future use other new APIs in reactor-core3.4.19+, I am wondering if we need always add this checking for that code. If this is only temporary, I am ok for it. 

And after users upgrade the SB version, are they better to upgrade the reactor-core version if they are using the old reactor-core and loading at runtime? we can mention that information in README.md. I am curious about the background of why they use the old reactor-core and load at runtime.

Reply:
Agree with Kun about the checking.

There is no background that users use old reactor-core I assume. If users don't need new feature for reactor-core or there is no bug in the past days, they may not upgrade it. So mention this information in README.md is enough.

Reply:
@anuchandy Looks like we need `ReactorShim`. Like we did for [jackson](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/ObjectMapperShim.java) 
@lmolkova might help here.

If we pursue this way, we should also make sure [this work item](https://github.com/Azure/azure-sdk-for-java/issues/28352) happens .

Reply:
Thanks Kamil, Kun & Zeija for the feedback.

Let me use ObjectMapperShim as a reference implementation to understand the design-pattern we established for cross-version handling. I'll follow up with Ludmila then.

Reply:
I was going over JacksonShim and ObjectMapperFactory; I feel like introducing `ReactorShim` in the azure-core for this case might be over-engineering. 

We're not changing the reactor-core minor or major version. Like other patch versions, there are no breaking changes in the "existing" operators. The case here is that _only_ Event Hubs SDK wants to use a "new" window operator overload from the latest patch, but if the user override the patch, it'll continue to use the "existing" window operator.

Today, if a user overrides with a different patch (3.4.patch) version, they won't face any linkage/method-not-found error; we continue to give the same guarantee.

So I feel we could keep it simple and limit to Event Hubs impl, Like cosmos SDK applying [warmup](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java#L52-L63) for reactor netty Client. At least that's what I'm thinking for the EH beta, which a customer is waiting to unblock their development.

Reply:
Sounds good, if EH is the only place that needs it let's keep it local.
However, I'd suggest to create `ReactorShim` in the EH `*.implementation.*` package so that we prepare ourselves for the eventual move to azure-core with this.

Reply:
Cool, Yes!. How is this [API-shape](https://github.com/Azure/azure-sdk-for-java/pull/29714/files#diff-52542f1b141e21042f2d808237c929f440f12b4209d4e239151ecf2187ecef54) look like?

Reply:
nice!